### PR TITLE
User Service Tests

### PR DESCRIPTION
**Pull request by githengi**

### CODE REVIEW

**Comment by keymane** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Clones added
============
- opensrp-app/src/test/java/org/smartregister/service/UserServiceTest.java  2
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/OpenSRP/opensrp-client-core/pullRequest?prid=5886005&bid=19267363)

**Comment by keymane** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Clones added
============
- opensrp-app/src/test/java/org/smartregister/service/UserServiceTest.java  4
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/OpenSRP/opensrp-client-core/pullRequest?prid=5886005&bid=19267363)

File: opensrp-app/src/test/java/org/smartregister/service/UserServiceTest.java [hunk 1]
```diff
@@ -300,21 +318,118 @@ public void testValidateDeviceTimeSameTimeTimeAndTimeZone() {
...
+        when(keyStore.containsAlias(user)).thenReturn(true);
+        KeyStore.PrivateKeyEntry privateKeyEntry = PowerMockito.mock(KeyStore.PrivateKeyEntry.class);
+        when(keyStore.getEntry(user, null)).thenReturn(privateKeyEntry);
+        String password = UUID.randomUUID().toString();
+        when(allSharedPreferences.fetchEncryptedPassword(user)).thenReturn(password);
+        when(allSharedPreferences.fetchEncryptedGroupId(user)).thenReturn(password);
+        assertFalse(userService.isUserInValidGroup(user, password));
+        verify(allSharedPreferences).fetchEncryptedPassword(user);
+        verify(allSharedPreferences, never()).fetchEncryptedGroupId(user);
+        verify(repository, never()).canUseThisPassword(password);
```
Comment (vincent-karuri) [inline opensrp-app/src/test/java/org/smartregister/service/UserServiceTest.java:385]: You could use ` verifyZeroInteractions` instead.

File: opensrp-app/src/test/java/org/smartregister/service/UserServiceTest.java [hunk 2]
```diff
@@ -300,21 +318,118 @@ public void testValidateDeviceTimeSameTimeTimeAndTimeZone() {
...
+    }
+
+    @Test
+    public void testGetGroupIdShShouldReturnNullOnError() throws Exception {
```
Comment (vincent-karuri) [inline opensrp-app/src/test/java/org/smartregister/service/UserServiceTest.java:389]: Typo?

**Pull request merged**